MariaDB/server
# MDEV-29582 deprecate mysql* names

- [x] *The Jira issue number for this PR is: MDEV-_____*


## Description

There are far too many old and new people in the community using MySQL and MariaDB interchangably which doesn't help our brand.

In help forums the use of mysql* names leave people abiguous on sources on correct information and misleads people assisting who don't recognise the mysql/mariadb status of the user.

In an effort to move user usage of mysql* names to mariadb* names that have been available since 10.4, we officially deprecate mysql* names with a warning and encourage the user to be MariaDB proud.

For those truely doing executable compatibility an environment variable MARIADB_PROUD=suppress_mysql_warnings enables that to happen without warnings.

mariadb-upgrade is adjusted to use mariadb-check and mariadb.

Pride also includes our test suite that now uses mariadb names.

## How can this PR be tested?

In addition to test suite:
```
$ client/mysql --help --verbose > /dev/null
client/mysql: Notice: client/mysql is deprecated, be MariaDB proud and use 'mariadb'

$ MARIADB_PROUD=suppress_mysql_warnings client/mysql --help --verbose > /dev/null
```



## Basing the PR against the correct MariaDB version
- [ X ] *This is a new feature and the PR is based against the latest MariaDB development branch*


## Backward compatibility

Scripts using old names gain an error in stdout. There's an environment variable for suppression if needed.
